protocol/messenger.go

```diff
@@ -3560,13 +3560,13 @@ func (r *ReceivedMessageState) addNewActivityCenterNotification(publicKey ecdsa.
 	}
 
 	if chat.CommunityChat() {
-		joinedClock, err := m.communitiesManager.GetCommunityRequestToJoinClock(&publicKey, message.CommunityID)
+		// Ignore mentions & replies in community before joining
+		joinedClock, err := m.communitiesManager.GetCommunityRequestToJoinClock(&publicKey, chat.CommunityID)
```

Comment:
is it possible to add a test for it?

Reply:
Yeah sure, it's work in progress at the moment

Reply:
Actually, there is already a test `TestAliceDoesNotReceiveCommunityNotificationsBeforeJoined'.

It doesn't fail even when I undo this PR commit. 
So my idea was to test the following scenario
1. After community is **created** and **announced** to Alice. She should not see any mentions from Bob
2. When she **spectates** the community, she should get a message, but the activity centre should have no notifications
3. And after **joining** the community, she should see all mentions in the activity centre.

```
	s.advertiseCommunityTo(community, bob, alice)
	_, err := alice.SpectateCommunity(community.ID())
	s.Require().NoError(err)

	mentionMessage := common.NewMessage()
	mentionMessage.ChatId = chat.ID
	mentionMessage.ContentType = protobuf.ChatMessage_TEXT_PLAIN
	mentionMessage.Text = "Good news, @" + common.EveryoneMentionTag + " !"

	response, err = bob.SendChatMessage(context.Background(), mentionMessage2)
	s.Require().NoError(err)

	response, err = WaitOnMessengerResponse(
		alice,
		func(r *MessengerResponse) bool {
			return len(r.Messages()) > 0
		},
		"no messages",
	)
	s.Require().NoError(err) <-- it doesn't receive any messages
```

Reply:
Updated PR

Reply:
One thing I don't understand and it's bothering me 

So, the test does the following:
1. Create a community with the default channel
2. Bob joins the community, alice is a spectator
3. Owner creates a new channel (!)
4. Bob sends a message with a mention to the default channel
5. Alice gets the message, but she has no notifications in the activity center
6. Alice joins the community, Bob sends another mention
7. Alice gets the message and an activity center notification

I don't understand why the CHAT_MESSAGE is not handled/delivered in RetrieveAll if I skip step 3

Reply:
I've fixed the test.
I just moved Bob to join the community after Alice starts spectating. And Alice waits for the community to be updated.

When Bob calls `joinCommunity`, it eventually calls `SaveAndPublish`, and on the Alice side it updates the community `persistence.SaveCommunity`.